frontends/web/src/routes/buy/pocket.tsx

```diff
@@ -152,6 +102,113 @@ export const Pocket = ({ code }: TProps) => {
     current.contentWindow?.postMessage(message, '*');
   };
 
+  const handleRequestAddress = (message: RequestAddressV0Message) => {
+    signing = true;
+    const addressType = message.withScriptType ? String(message.withScriptType) : '';
+    const withMessageSignature = message.withMessageSignature ? String(message.withMessageSignature) : '';
+    signAddress(
+      addressType,
+      withMessageSignature,
+      code)
+      .then(response => {
+        signing = false;
+        if (response.success) {
+          sendAddress(response.address, response.signature);
+        } else {
+          if (response.errorCode !== 'userAbort') {
+            alertUser(t('genericError'));
+            console.log('error: ' + response.errorMessage);
+          }
+        }
+      });
+
+  };
+
+  const handleVerifyAddress = (address: string) => {
+    setVerifying(true);
+    verifyAddress(address, code)
+      .then(response => {
+        setVerifying(false);
+        if (!response.success) {
+          if (response.errorCode === 'addressNotFound') {
+            // This should not happen, unless the user receives a tx on the same address between the message signing
+            // and the address verification.
+            alertUser(t('buy.pocket.usedAddress', { address:  address }));
+          } else {
+            alertUser(t('genericError'));
+            console.log('error: ' + response.errorMessage);
+          }
+        }
+      });
+  };
+
+  const sendXpub = () => {
+    if (accountInfo) {
+      let bitcoinSimple = accountInfo.signingConfigurations[0].bitcoinSimple;
+      if (bitcoinSimple) {
+        const xpub = bitcoinSimple.keyInfo.xpub;
+        const { current } = iframeRef;
+        if (!current) {
+          return;
+        }
+        const message = serializeMessage({
+          version: MessageVersion.V0,
+          type: V0MessageType.ExtendedPublicKey,
+          extendedPublicKey: xpub,
+        });
+        current.contentWindow?.postMessage(message, '*');
+      }
+    }
+  };
+
+  const handleRequestXpub = () => {
+    getTransactionList(code as AccountCode).then(txs => {
+      if (txs.length > 0) {
```

Comment:
I think this is fine: but getTransactionList gets all transactions (unified accounts). Only the SegWit (p2wpkh) xpub will be shared. So user could have nothing on p2wpkh account, but we still show this warning if there is coins on p2tr or p2wpkh-p2sh.
As mentioned I think it is good as is.

Reply:
also is `as AccountCode` needed?

I'll put on my todo to look into if we should type code everywhere to use AccountCode instead of string.

Reply:
> I think this is fine: but getTransactionList gets all transactions (unified accounts). Only the SegWit (p2wpkh) xpub will be shared. So user could have nothing on p2wpkh account, but we still show this warning if there is coins on p2tr or p2wpkh-p2sh.
As mentioned I think it is good as is.

You're right! this is something to keep in mind, but I also think it is good as is.